recipes-bsp/bootfiles/rpi-config.bb

```diff
@@ -2,18 +2,16 @@ DESCRIPTION = "Commented config.txt file for the Raspberry Pi. \
                The Raspberry Pi config.txt file is read by the GPU before \
                the ARM core is initialised. It can be used to set various \
                system configuration parameters."
+HOMEPAGE = "https://www.raspberrypi.com/documentation/computers/config_txt.html"
```

Comment:
I'm not sure what to say about this commit. I feel like users are used to the format and will end up with a pretty confusing change. @kraj what do you think?

Reply:
I really don't think that a commented out file was adding any value to the user. On the contrary, this was just cluttered and error prone. The only useful addition to the config file would be to echo the homepage URL as the first line so that the user knows where to look for documentation if they want to tweak the config, not scrolling a subjectively commented out file.

Reply:
I see both sides - I'm unsure how to proceed.;

Reply:
@kraj your thoughts on this?

Reply:
config.txt has detailed comments for an option, in-place sed keeps that intact. Non yocto users are familiar with this as well, since thats the first place people go to debug weird issues. So keeping the format has advantages. Sed is as reliable as echo, both are shell commands. They can fail in same ways. If sed can go wrong so can echo emit an older option and render system problems. So I am not buying echo is better than sed argument :)

Reply:
how do we handle the fact that our config.txt mechanism is our of sync with upstream config.txt, does this happen ?
do we then point them to wrong config.txt as reference ? Secondly, I think lot of users edit it in place to debug things so a symlink at top might be fine to point them to what these options mean maybe

Reply:
I'm not sure to understand what you mean by upstream config.txt, reference config.txt or symlink. I also don't understand how a file fetched from some random user repository on Github tackles any of the issues you're trying to describe here. The cleaner the file is, the less error-prone it gets.

Reply:
the original `config.txt` that we will overwrite if we accept this PR.

Reply:
I want to clarify that there isn't an "original" config.txt file that we are overwriting with this PR. The initial recipe fetched an [unofficial file](https://github.com/Evilpaul/RPi-config) from GitHub.

Different distributions come with their own config.txt files. For instance, RPI-Distro (formerly Raspbian) ships [this one](https://raw.githubusercontent.com/RPi-Distro/pi-gen/master/stage1/00-boot-files/files/config.txt), while Arch Linux ARM ships [this version](https://github.com/archlinuxarm/PKGBUILDs/blob/master/core/linux-rpi/config.txt). The choice of config.txt is determined by the distribution, not the recipe.

In fact, a minimal image can successfully boot with either no config.txt file or with the following content:

```
# Empty file
# Some settings may impact device functionality. Refer to the official documentation below for details.
# https://www.raspberrypi.com/documentation/computers/config_txt.html
```

As a BSP layer, our goal should be to provide a clean default configuration with minimal adjustments. Distributions have the flexibility to further customize the configuration using conditional filters, additional files like autoboot.txt, or any configuration fragments enabled with the include directive, as outlined in the documentation.

Following best practices, it would be advisable to list rpi-config in MACHINE_EXTRA_RDEPENDS to prevent core-image-minimal from installing the file by default, while allowing other core images to include it. In the meantime, I kindly request that @kraj and @agherzan carefully consider this comment and refrain from advocating for the use of an unofficial text file on the internet, as it may not align with established best practices.

Reply:
I would recommend this being split in another PR. It does not directly support the camera modules and we can have more public visibility on this in this way.

Reply:
I would personally prefer a minimal config.txt that isn't filled with every option imaginable.

Reply:
@vivien I started to lean into that, too - with the reoccurrence of the config.txt size issues in RPI5. @kraj I think we should start considering this as a cleaner solution forward.